src/redux-reducer-effects.ts

```diff
@@ -51,32 +67,49 @@ const enhance = (originalCreateStore: StoreCreator) => {
 }
 export default enhance;
 
+type Accumulator<S,Task> = {
+    state: S,
+    tasks: Task[],
+};
+
+
 type Dictionary<T> = { [index: string]: T; }
 export const combineReducers = <S, Task>(reducerMap: EnhancedReducersMapObject<Task>): EnhancedReducer<S, Task> => {
-    return <Msg>(state: Dictionary<any>, msg: Msg): [S, Option<Task[]>] => {
-        type Accumulator = {
-            state: any,
-            tasks: Option<Task[]>[]
-            hasChanged: boolean
-        };
-        const model = Object.keys(reducerMap).reduce<Accumulator>((acc, key) => {
+    return <Msg>(state: Dictionary<any>, msg: Msg): [S, Task[]] => {
+        const model = Object.keys(reducerMap).reduce<Accumulator<any,Task>>((acc, key) => {
             const reducer = reducerMap[key];
+
             // We lose type safety here because state is a record
             const previousStateForKey = state[key];
-            const nextResultForKey = reducer(previousStateForKey, msg);
-            const [nextStateForKey, maybeTasks] = nextResultForKey;
+            const result = <EnhancedReducerResult<S,Task>>reducer(previousStateForKey, msg);
+            acc.state[key] = getState(result);
+            acc.tasks.push(...getTasks(result));
 
-            acc.hasChanged = acc.hasChanged || nextStateForKey !== previousStateForKey;
-            acc.state[key] = nextStateForKey;
-            acc.tasks.push(maybeTasks);
             return acc;
         }, {
             state: {},
             tasks: [],
-            hasChanged: false
         });
 
-        const tasks = (<Task[]>[]).concat(...(flattenOptions(model.tasks)));
-        return [model.state, new Some(tasks)];
+        return [model.state, model.tasks];
     };
 }
+
+export const composeReducers = <S, Task>(...reducers: EnhancedReducer<S, Task>[]): EnhancedReducer<S, Task> => {
+  return <Msg>(state: S, msg: Msg): [S, Task[]] => {
+    const model = reducers.reduce<Accumulator<S,Task>>((acc, reducer) => {
+        const result = reducer(acc.state, msg);
+
+        acc.state = getState(result);
+        acc.tasks.push(...getTasks(result));
+
+        return acc;
```

Comment:
This reducer function looks very similar to the one used for `combineReducers`. I wonder if they could share the same inner reducer function?